### DISCUSSION TITLE
Kubernetes 1.30 Major Themes

### DISCUSSION THREAD

**Discussion by rashansmith** [kubernetes/sig-release#2424] [General]
This is a discussion to discuss Major Themes for the v1.30 release.

Here is the [discussion thread](https://github.com/kubernetes/sig-release/discussions/2349) from the last release cycle.

cc @kubernetes/release-team-leads
cc @fykaa @npolshakova @OrlinVasilev @satyampsoni

**Comment by knabben** [comment]
SIG-Windows:

- [Node Log Query](https://github.com/kubernetes/enhancements/pull/4480) graduates to Beta in 1.30
- [Windows Operational Readiness](https://github.com/kubernetes-sigs/windows-operational-readiness/releases/tag/0.1.0) - First release v0.1.0

**Reply by marosset** [reply]
And memory pressure eviction support (assuming https://github.com/kubernetes/kubernetes/pull/122922 merges in this release)

**Comment by xing-yang** [comment]
From SIG-Storage:

- [Robust VolumeManager Reconstruction](https://github.com/kubernetes/enhancements/issues/3756) feature moved to GA in v1.30. This feature improves Kubelet handling of mounted volumes across restarts, enabling faster, more robust recovery.
- [Prevent Unauthorized Volume Mode Conversion](https://github.com/kubernetes/enhancements/issues/3141) feature moved to GA in v1.30. This feature prevents unauthorized conversion of the volume mode when creating a Persistent Volume Claim from a Volume Snapshot. The prevent-volume-mode-conversion feature flag is enabled by default in external-snapshotter v7.0.0+ and external-provisioner v4.0.0+. Note that this is a breaking change. Volume mode change will be rejected when creating a PVC from a VolumeSnapshot unless the AllowVolumeModeChange annotation has been set to true. Applications relying on volume mode change when creating a PVC from VolumeSnapshot need to be updated accordingly.
- [Speed up SELinux Volume Relabeling using Mounts](https://github.com/kubernetes/enhancements/issues/1710) feature moved to alpha in v1.30 for all PV access modes (previously moved to Beta in v1.27 for ReadWriteOncePod PV access mode). This feature speeds up container startup by mounting volumes with the correct SELInux label instead of recursively changing each file on the volume.
cc @saad-ali @msau42 @jsafrane

**Comment by gjtempleton** [comment]
SIG-Autoscaling:
* [Container resource based horizontal autoscaling](https://github.com/kubernetes/enhancements/issues/1610) graduates to GA in v1.30

**Comment by kerthcet** [comment]
FROM SIG-Scheduling:
* [Pod Scheduling Readiness](https://github.com/kubernetes/enhancements/issues/3521) moved to GA
* [Min domains in PodTopologySpread](https://github.com/kubernetes/enhancements/issues/3022) moved to GA

cc @kubernetes/sig-scheduling-misc

**Comment by aojea** [comment]
SIG Network:
* [KEP-4444: Routing Preference for Services](https://github.com/kubernetes/enhancements/issues/4444) new alpha feature introducing a new field, `routingPreference`, to the Kubernetes Service spec. It will superseding the functionality currently provided by the service.kubernetes.io/topology-mode annotation.
* [Make Kubernetes aware of the LoadBalancer behaviour #1860](https://github.com/kubernetes/enhancements/issues/1860) moves to beta

**Comment by tenzen-y** [comment]
From SIG Apps / WG Batch:

- [KEP-3998 Job success/completion policy](https://github.com/kubernetes/enhancements/issues/3998): New alpha feature to define when a Job can be declared as succeeded based on the number of succeeded pods. 
- [KEP-4368 Job API managed-by mechanism](https://github.com/kubernetes/enhancements/issues/4368): New alpha feature to delegate reconciliation of a Job object to an external controller.

@soltysh @atiratree @alculquicondor @mimowo @danielvegamyhre @ahg-g @kannon92 Is there any major themes?

**Reply by alculquicondor** [reply]
LGTM from the job side of things
